hibernate-core/src/main/java/org/hibernate/query/sqm/tree/select/SqmSortSpecification.java

```diff
@@ -16,22 +16,21 @@
  * @author Steve Ebersole
  */
 public class SqmSortSpecification implements JpaOrder {
-	private SqmExpression sortExpression;
-	private String collation;
-	private SortOrder sortOrder;
-	private final NullPrecedence nullPrecedence;
```

Comment:
please note: `nullPrecenence` duplicates `precedence` field.
`nullPrecendence` was assigned only in constructor and **never used**, all other access is done to `precedence` field.
I looks like a defect. I fixed it, because otherwise I could not write a test to validate all assignments in this class.

Reply:
I prefer the name `nullPrecedence` - its more consistent and self-documenting

Reply:
fixed